ante/fee.go

```diff
@@ -93,3 +162,54 @@ func GetTxPriority(fee sdk.Coins) int64 {
 
 	return priority
 }
+
+// overwrite DenomsSubsetOf from sdk, to allow zero amt coins. e.g. 1stake is DenomsSubsetOf 0stake.
+// DenomsSubsetOf returns true if coins's denoms is subset of coinsB's denoms.
+// if coins is empty set, empty set is any sets' subset
+func DenomsSubsetOf(coins, coinsB sdk.Coins) bool {
+	for _, coin := range coins {
+		// validate denom
+		err := sdk.ValidateDenom(coin.Denom)
+		if err != nil {
+			panic(err)
+		}
+
+		if ok, _ := coinsB.Find(coin.Denom); !ok {
+			return false
+		}
+	}
+
+	return true
+}
+
+// overwrite the IsAnyGTE from sdk to allow zero coins.
+// IsAnyGTE returns true iff coins contains at least one denom that is present
+// at a greater or equal amount in coinsB; it returns false otherwise.
+// NOTE: IsAnyGTE operates under the invariant that both coin sets are sorted
+// by denominations.
+func IsAnyGTE(coins, coinsB sdk.Coins) bool {
+	// this is different from sdk, sdk return false
+	if len(coinsB) == 0 {
+		// global fee should not be not empty
+		panic("empty coin set")
+	}
+	// if feecoins empty, and globalfee has one denom of amt zero. feecoins equals to that 0denom.
```

Comment:
if `len(coins) == 0` shouldn't it just pass?
Replacing `coins` with `coinsB` means coins will always be GTE to `coinsB`, right?